tilequeue/query/rawr.py

```diff
@@ -0,0 +1,578 @@
+from collections import namedtuple, defaultdict
+from shapely.geometry import box
+from shapely.wkb import loads as wkb_loads
+from tilequeue.query.common import layer_properties
+from tilequeue.query.common import is_station_or_stop
+from tilequeue.query.common import is_station_or_line
+from tilequeue.query.common import deassoc
+from tilequeue.query.common import mz_is_interesting_transit_relation
+from tilequeue.query.common import shape_type_lookup
+from tilequeue.transform import calculate_padded_bounds
+from raw_tiles.tile import shape_tile_coverage
+from math import floor
+
+
+class Relation(object):
+    """
+    Relation object holds data about a relation and provides a nicer interface
+    than the raw tuple by turning the tags array into a dict, and separating
+    out the "parts" array of IDs into separate lists for nodes, ways and other
+    relations.
+    """
+
+    def __init__(self, rel_id, way_off, rel_off, parts, members, tags):
+        self.id = rel_id
+        self.tags = deassoc(tags)
+        self.node_ids = parts[0:way_off]
+        self.way_ids = parts[way_off:rel_off]
+        self.rel_ids = parts[rel_off:]
+
+
+class TilePyramid(namedtuple('TilePyramid', 'z x y max_z')):
+    """
+    Represents a "tile pyramid" of all tiles which are geographically
+    contained within the tile `z/x/y` up to a maximum zoom of `max_z`. This is
+    the set of tiles corresponding to one RAWR tile.
+    """
+
+    def tile(self):
+        from raw_tiles.tile import Tile
+        return Tile(self.z, self.x, self.y)
```

Comment:
I wonder if we should at some point just create our own repo to hold these kinds of objects and related utility functions?

Reply:
Follow-up in #246.